modules/geo-layers/src/imagery-layer/imagery-layer.ts

```diff
@@ -0,0 +1,205 @@
+          (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged));
+
+
+      // Check if data source has changed
+      if (dataChanged) {
+        state.imageSource = createImageSource(this.props);
+        this._loadMetadata();
+        debounce(() => this.loadImage('image source changed'), 0);
+      }
+
+      // Some sublayer props may have changed
+    }
+    
+    if (changeFlags.viewportChanged) {
+      debounce(() => this.loadImage('viewport changed'));
+    }
+
+    const propsChanged = changeFlags.propsOrDataChanged || changeFlags.updateTriggersChanged;
+  }
+  
+  /*override*/ finalizeState(): void {
+    // TODO - we could cancel outstanding requests
+  }
+  
+  /*override*/ renderLayers(): Layer {
+    // TODO - which bitmap layer is rendered should depend on the current viewport
+    // Currently Studio only uses one viewport
+    const state = this.state as ImageryLayerState;
+    const {imageSource} = this.state;
+    const {bounds, image, width, height} = this.state;
+
+    console.log('render', this.state);
+
+    return image && new BitmapLayer({
+      ...this.getSubLayerProps({id: 'bitmap'}),
+      bounds,
+      image,
+      pickable: true, // TODO inherited?
+      onHover: (info) => console.log('hover in bitmap layer', info),
+      onClick: (info) => {
+        // @ts-expect-error
+        const {bitmap, layer} = info;
+        console.log('click in bitmap layer', info);
+        if (bitmap) {
+          const x = bitmap.pixel[0];
+          const y = bitmap.pixel[1];
+          debounce(async () => {
+            const featureInfo = await imageSource.getFeatureInfo({
+              layers: this.props.layers,
+              // todo image width may get out of sync with viewport width
+              width,
+              height,
+              bbox: bounds,
+              query_layers: this.props.layers,
+              x,
+              y,
+              info_format: 'text/plain'
+            })
+            // console.log(featureInfo);
+          }, 0);
+        }
+      }
+    });
+  }
+
+  /** Run a getMetadata on the image service */
+  async _loadMetadata(): Promise<void> {
+    this.props.onMetadataLoadStart();
+    const state = this.state as ImageryLayerState;
+    try {
+      state.metadata = await state.imageSource.getMetadata();
+      // technically we should get the latest layer after an async operation in case props have changed
+      // Although the response might no longer be expected
+      this.getCurrentLayer()?.props.onMetadataLoadComplete(state.metadata);
+    } catch (error) {
+      this.getCurrentLayer()?.props.onMetadataLoadError(error as Error);
+    }
+  }
+  
+  /** Load an image */
+  async loadImage(reason: string): Promise<void> {
+    const viewports = this.context.deck?.getViewports() || [];
+    if (viewports.length <= 0) {
+      return;
+    }
+
+    // TODO - need to handle multiple viewports
+    const viewport = viewports[0];
```

Comment:
This may not be the correct viewport. Pass `this.context.viewport` from `updateState` when you call this method.

We have a handful of layers that cannot render in multiple views simultaneously, but at the minimum, a layer should render correctly if it is bound to a single view by `layerFilter`. See https://deck.gl/docs/developer-guide/views#rendering-layers-in-multiple-views

Reply:
Using `context.viewport`